### PR TITLE
Check if roles with hyphen work with walrus

### PR DESCRIPTION
**Pull request by olirice**
## What kind of change does this PR introduce?
Add test for role with a hyphen

### CODE REVIEW

File: test/expected/test_role_with_hyphen.out [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ ?column? 
+----------
+        1
+(1 row)
+
+create role "has-hyphen" nologin noinherit;
+grant usage on schema public to "has-hyphen";
+alter default privileges in schema public grant all on tables to "has-hyphen";
+alter default privileges in schema public grant all on functions to "has-hyphen";
+alter default privileges in schema public grant all on sequences to "has-hyphen";
```
Comment (olirice) [inline test/expected/test_role_with_hyphen.out:11]: new role created here

File: test/expected/test_role_with_hyphen.out [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+create table public.notes(
+    id int primary key
+);
+insert into realtime.subscription(subscription_id, entity, claims)
+select
+    seed_uuid(1),
+    'public.notes',
+    jsonb_build_object(
+        'role', 'has-hyphen',
```
Comment (olirice) [inline test/expected/test_role_with_hyphen.out:20]: new role referenced here

File: test/expected/test_role_with_hyphen.out [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+     "schema": "public",                           +|                |                                        | 
+     "columns": [                                  +|                |                                        | 
+         {                                         +|                |                                        | 
+             "name": "id",                         +|                |                                        | 
+             "type": "int4"                        +|                |                                        | 
+         }                                         +|                |                                        | 
+     ],                                            +|                |                                        | 
+     "commit_timestamp": "2000-01-01T08:01:01.000Z"+|                |                                        | 
+ }                                                  |                |                                        | 
+(1 row)
```
Comment (olirice) [inline test/expected/test_role_with_hyphen.out:55]: output is correct, permissions check passed

File: sql/walrus_migration_0010_quoted_roles.sql [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+            ) loop
+
+                if not is_rls_enabled or action = 'DELETE' then
+                    visible_to_subscription_ids = visible_to_subscription_ids || subscription_id;
+                else
+                    -- Check if RLS allows the role to see the record
+                    perform
+                        -- Trim leading and trailing quotes from working_role because set_config
+                        -- doesn't recognize the role as valid if they are included
+                        set_config('role', trim(both '"' from working_role::text), true),
```
Comment (olirice) [inline sql/walrus_migration_0010_quoted_roles.sql:270]: this is the change. trimming possible `"` from the front and back after a `regrole` gets cast to text.
I looked for tidier cleaner solution but there isn't one

File: test/expected/test_role_with_hyphen.out [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+drop table private.notes;
+drop schema private;
+select pg_drop_replication_slot('realtime');
+ pg_drop_replication_slot 
+--------------------------
+ 
+(1 row)
+
+truncate table realtime.subscription;
```
Comment (olirice) [inline test/expected/test_role_with_hyphen.out:71]: full test output

File: test/expected/test_subscribers_multiple_roles.out
Comment (olirice) [inline test/expected/test_subscribers_multiple_roles.out:1]: this file didn't actually change. The two records are now in reversed order

**Review by w3b6x9 — APPROVED** [review]
lgtm, thanks!

**Pull request merged**